[CI] EsqlSpecIT test {lookup-join.LookupMessageFromIndexKeepReordered SYNC} failing

**Build Scans:**
- [elasticsearch-intake #14014 / part3](https://gradle-enterprise.elastic.co/s/rczqx36rb4dui)
- [elasticsearch-intake #13961 / part3](https://gradle-enterprise.elastic.co/s/el7odw3ww3sco)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT" -Dtests.method="test {lookup-join.LookupMessageFromIndexKeepReordered SYNC}" -Dtests.seed=53BC5891FD5B4194 -Dtests.locale=vai-LR -Dtests.timezone=Pacific/Chuuk -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Blookup-join.LookupMessageFromIndexKeepReordered%20SYNC%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Data mismatch:
row 0 column 0:a list containing
row 0 column 0:0: expected "Success" but was "Disconnected"
row 0 column 1:a list containing
row 0 column 1:0: expected "172.21.3.15" but was "172.21.0.5"
row 0 column 2:a list containing
row 0 column 2:0: expected "1756467" but was "1232382"
row 0 column 3:a list containing
row 0 column 3:0: expected "Connected to 10.1.0.1" but was "Disconnected"
row 1 column 0:a list containing
row 1 column 0:0: expected "Error" but was "Success"
row 1 column 1:a list containing
row 1 column 1:0: expected "172.21.3.15" but was "172.21.2.113"
row 1 column 2:a list containing
row 1 column 2:0: expected "5033755" but was "2764889"
row 1 column 3:a list containing
row 1 column 3:0: expected "Connection error" but was "Connected to 10.1.0.2"
row 2 column 0:a list containing
row 2 column 0:0: expected "Error" but was "Success"
row 2 column 1:a list containing
row 2 column 1:0: expected "172.21.3.15" but was "172.21.2.162"
row 2 column 2:a list containing
row 2 column 2:0: expected "8268153" but was "3450233"
```

**Issue Reasons:**
- [main] 2 failures in test test {lookup-join.LookupMessageFromIndexKeepReordered SYNC} (3.2% fail rate in 62 executions)
- [main] 2 failures in step part3 (14.3% fail rate in 14 executions)
- [main] 2 failures in pipeline elasticsearch-intake (14.3% fail rate in 14 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -245,12 +245,6 @@ tests:
 - class: org.elasticsearch.packaging.test.ArchiveTests
   method: test41AutoconfigurationNotTriggeredWhenNodeCannotContainData
   issue: https://github.com/elastic/elasticsearch/issues/118110
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {lookup-join.LookupMessageFromIndexKeepReordered SYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/118150
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {lookup-join.LookupMessageFromIndexKeepReordered ASYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/118151
 - class: org.elasticsearch.xpack.remotecluster.CrossClusterEsqlRCS2UnavailableRemotesIT
   method: testEsqlRcs2UnavailableRemoteScenarios
   issue: https://github.com/elastic/elasticsearch/issues/117419

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/lookup-join.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/lookup-join.csv-spec
@@ -283,6 +283,7 @@ FROM sample_data
 | LOOKUP JOIN message_types_lookup ON message
 | KEEP type, client_ip, event_duration, message
 ;
+ignoreOrder:true
 
 type:keyword | client_ip:ip | event_duration:long | message:keyword
 Success      | 172.21.3.15  | 1756467             | Connected to 10.1.0.1

